crates/iceberg/src/io.rs

```diff
@@ -278,7 +288,7 @@ impl OutputFile {
     }
 
     /// Creates output file for writing.
-    pub async fn writer(&self) -> Result<impl FileWrite> {
+    pub async fn writer(&self) -> Result<Writer> {
```

Comment:
This interface looks weird because returning a type here can't let us determine the type in `ParquetWriter`. So I modify it to a specific type here temporarily. 

I think if we want to make it generic, maybe we need to make `OutputFile` a trait and let OutputFile bind an associate type like🤔
```
trait OutputFile {
   type writer: FileWriter;
   ...
}
```

cc @liurenjie1024

Reply:
We can't make it concrete `Writer` here since we can't expose opendal api to user. You can store `Box<dyn FileWriter>` in parquet writer.